src/vtcomposite.cpp

```diff
@@ -826,104 +824,130 @@ struct LocalizeWorker : Napi::AsyncWorker
+    Napi::Object params = info[0].As<Napi::Object>();
+
+    // params.buffer (required)
+    if (!params.Has(Napi::String::New(info.Env(), "buffer")))
     {
-        return utils::CallbackError("language value must be null or a string", info);
+        return utils::CallbackError("params.buffer is required", info);
     }
-    else
+    Napi::Value buffer_val = params.Get(Napi::String::New(info.Env(), "buffer"));
+    if (!buffer_val.IsObject() || buffer_val.IsNull() || buffer_val.IsUndefined())
     {
-        language = language_val.As<Napi::String>();
+        return utils::CallbackError("params.buffer must be a Buffer", info);
     }
 
-    if (change_names && language.length() == 0)
+    Napi::Object buffer_obj = buffer_val.As<Napi::Object>();
+    if (!buffer_obj.IsBuffer())
     {
-        return utils::CallbackError("language value is an empty string", info);
+        return utils::CallbackError("params.buffer is not a true Buffer", info);
     }
+    buffer = buffer_obj.As<Napi::Buffer<char>>();
 
-    // validate worldview string
-    Napi::Value worldview_val = info[2];
-    std::string worldview;
-
-    if (!worldview_val.IsString() && !worldview_val.IsNull())
+    // params.language (optional)
+    if (params.Has(Napi::String::New(info.Env(), "language")))
     {
-        return utils::CallbackError("worldview value must be null or a 2 character string", info);
+        Napi::Value language_val = params.Get(Napi::String::New(info.Env(), "language"));
+        if (!language_val.IsString() && !language_val.IsNull() && !language_val.IsUndefined())
+        {
+            return utils::CallbackError("params.language must be null or a string", info);
+        }
+        if (language_val.IsString())
+        {
+            language = language_val.As<Napi::String>();
+            if (language.length() == 0)
+            {
+                return utils::CallbackError("params.language cannot be an empty string", info);
+            }
+        }
+        if (language_val.IsNull() || language_val.IsUndefined())
+        {
+            change_names = false;
+        }
+    }
+    else
+    {
+        change_names = false;
     }
 
-    if (worldview_val.IsString())
+    // params.worldview
+    if (params.Has(Napi::String::New(info.Env(), "worldview")))
     {
-        worldview = worldview_val.As<Napi::String>();
-        if (worldview.length() != 2)
+        Napi::Value worldview_val = params.Get(Napi::String::New(info.Env(), "worldview"));
+        if (!worldview_val.IsArray())
         {
-            return utils::CallbackError("worldview must be a string 2 characters long", info);
+            return utils::CallbackError("params.worldview must be an array", info);
+        }
+        Napi::Array worldview_array = worldview_val.As<Napi::Array>();
+        std::uint32_t num_worldviews = worldview_array.Length();
+        worldview.reserve(num_worldviews);
+        for (std::uint32_t wv = 0; wv < num_worldviews; ++wv)
```

Comment:
same, I think this would be clearar as `auto wv = 0 ;...`

Reply:
Napi::Array `Length()` returns an unsigned int https://github.com/nodejs/node-addon-api/blob/main/doc/array.md#length, so when comparing an `auto wv = 0` to `num_worldviews` we get the following compile errors: 

```cpp
../src/vtcomposite.cpp:914:66: error: implicit conversion changes signedness: 'int' to 'uint32_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
            Napi::Value worldview_item_val = worldview_array.Get(wv);
                                                             ~~~ ^~
../src/vtcomposite.cpp:912:30: error: comparison of integers of different signs: 'int' and 'unsigned int' [-Werror,-Wsign-compare]
        for (auto wv = 0; wv < num_worldviews; ++wv)
                          ~~ ^ ~~~~~~~~~~~~~~
2 errors generated.
```

IMO this doesn't feel clearer than explicitly typing the variables.